### PR TITLE
GH-33517: [C++][Flight] Exercise UCX on CI

### LINKED ISSUE #33517: [CI][C++][Flight] Exercise UCX on CI
UCX doesn't seem enabled on any CI configuration for now.

We should have at least a nightly job with UCX enabled, for example one of the Conda or Ubuntu builds.

**Reporter**: [Antoine Pitrou](https://issues.apache.org/jira/browse/ARROW-18349) / @pitrou
#### PRs and other links:
- [GitHub Pull Request #14667](https://github.com/apache/arrow/pull/14667)

<sub>**Note**: *This issue was originally created as [ARROW-18349](https://issues.apache.org/jira/browse/ARROW-18349). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

### PR DESCRIPTION
**Pull request by pitrou**
* Closes: #33517

### CODE REVIEW

**Comment by pitrou** [comment]
@lidavidm

**Comment by lidavidm** [comment]
LGTM, though guess I'll have to go look into the crashes/hangs first. Thanks for looking at this.

**Comment by pitrou** [comment]
@lidavidm I rebased and the UCX tests passed here:
https://github.com/apache/arrow/actions/runs/4993639063/jobs/8943049374?pr=14667

**Review by lidavidm — APPROVED** [review]
Thanks!

**Review by assignUser — APPROVED** [review]
+1

File: ci/docker/conda-cpp.dockerfile [hunk 1]
```diff
@@ -36,6 +36,7 @@ RUN mamba install -q -y \
         compilers \
         doxygen \
         python=${python} \
+        ucx-proc=*=cpu ucx libnuma \
```
Comment (kou) [inline ci/docker/conda-cpp.dockerfile:39]: How about putting one package per line like others?

```suggestion
        libnuma \
        python=${python} \
        ucx \
        ucx-proc=*=cpu \
```

**Review by kou — APPROVED** [review]
+1

**Pull request merged**